File: Smartsense.Mobile.Android/app/src/main/java/com/smartsense/covid/api/model/responses/BaseApiResponse.java
package com.smartsense.covid.api.model.responses;

import java.util.ArrayList;
import java.util.List;

public class BaseApiResponse {
    private String code;
    private String message;
    private List<String> errors;

    public BaseApiResponse(List<String> errors) {
        this.errors = new ArrayList<>();
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public List<String> getErrors() {
        return errors;
    }

    public void setErrors(List<String> errors) {
        this.errors = errors;
    }
}

File: Smartsense.Mobile.iOS/covid/covid/Api/Responses/BaseApiResponse.swift
//
//  BaseApiResponse.swift
//  covid
//
//  Created by SmartSense on 8.06.2021.
//

import Foundation

protocol BaseApiResponse {
    var code: String?  { get set }
    var message: String?  { get set }
    var errors: [String?]?  { get set }
}
